Setup updates for Linux (#888)
o3de/o3de.org: content/docs/welcome-guide/requirements.md

section, each prerequisite is listed along with any specific configuration steps
required.

> # Hardware requirements

Reviewer: It goes against the doc style to use an H1 header for these subsections. However, looking at the deploy preview, I think I see why you did this: the most useful links remain in the right-hand ToC after adding the new sections.

Instead of using H1 headers though, I'd rather keep them at H2, while keeping **Microsoft Windows** and the remaining OS sections at H2.
```suggestion
## Hardware requirements
```

{{< /note >}}

> # Software prerequisites and configuration {#software-prerequisites}

Reviewer: ```suggestion
## Software prerequisites and configuration {#software-prerequisites}
```

{{< note >}}
In order to perform the download step shown below, you must install [Curl](https://curl.se):
```shell
sudo apt-get install curl
```
{{< /note >}}

```shell
pushd /tmp

LIBFFI_PACKAGE_NAME=libffi6_3.2.1-8_amd64.deb
LIBFFI_PACKAGE_URL=http://mirrors.kernel.org/ubuntu/pool/main/libf/libffi/

curl --location $LIBFFI_PACKAGE_URL/$LIBFFI_PACKAGE_NAME -o $LIBFFI_PACKAGE_NAME

sudo apt install ./$LIBFFI_PACKAGE_NAME

popd
```

### Additional library dependencies

O3DE also requires some additional library packages to be installed:

* libglu1-mesa-dev
* libxcb-xinerama0
* libxcb-xinput0
* libxcb-xkb-dev
* libxkbcommon-dev
* libxkbcommon-x11-dev
* libfontconfig1-dev
* libcurl4-openssl-dev                    
* libsdl2-dev                             
* zlib1g-dev
* mesa-common-dev

These packages can be downloaded and installed through `apt-get`.

```shell
sudo apt-get install libglu1-mesa-dev libxcb-xinerama0 libxcb-xinput0 libfontconfig1-dev libcurl4-openssl-dev libsdl2-dev zlib1g-dev zlib1g-dev mesa-common-dev libxcb-xkb-dev libxkbcommon-dev libxkbcommon-x11-dev curl
```

### Ninja Build System (Optional)

> O3DE supports multiple build configurations (debug, profile, and release), and you will need to specify which configuration you want to use when building O3DE. By default, CMake supports Linux's default Unix Makefiles; however, Unix Makefiles only supports single-configuration builds and not multi-configuration builds. That means you must determine which configuration you want to use for the build when you generate the project. All builds for the project will be based on that configuration. In order to change the build configuration, you will need to regenerate the project with the different configuration.
> 
> With the Ninja build system, specifically [Ninja Multi-Config](https://cmake.org/cmake/help/latest/generator/Ninja%20Multi-Config.html), you will be able to generate the project once but can determine which configuration to build during build time (see the [Linux setup guide](/docs/user-guide/platforms/linux) for more details). We recommend that you use this generator for O3DE development. You can install Ninja built tool through `apt-get`. 

Reviewer 2: - Simplified the first sentence. Then, moved it as the second sentence because it provides additional information, rather than serving as an introductory sentence.
- Simplified verb phrases in sentences (e.g. "will be able to" to "can")
- Added the last sentence in the 1st paragraph to summarize the limitation with Unix Makefiles. 
- Added the first sentence to the 2nd paragraph to introduce the Ninja build system and its relationship to the Unix Makefile build system. This clarifies that Ninja is used as an alternative to Unix Makefile, rather than in conjunction. 

```suggestion
By default, CMake uses Unix Makefiles for building O3DE. O3DE supports multiple build configurations (debug, profile, and release), which you specify when building O3DE. Unix Makefiles only supports single-configuration builds, so you must determine which configuration you want to build when you generate the project. All project builds are based on that configuration. In order to change the build configuration, you need to regenerate the project with the different configuration. This process restricts O3DE's support for multiple-configuration builds and slows down building workflows.

The Ninja build system is an alternative to Linux's default Unix Makefiles that makes it easier to generate, configure, and build your project. With the Ninja build system, specifically [Ninja Multi-Config](https://cmake.org/cmake/help/latest/generator/Ninja%20Multi-Config.html), you can generate the project once and determine which configuration to build during build time. We recommend that you use this generator for O3DE development. You can install Ninja built tool through `apt-get`. 
```

1. Choose **OK** to save your changes and close the windows that were opened.

1. Verify that `cmake` is on the system path by opening a new command line window and running `cmake --version` again.

## macOS

Support for developing on macOS is in an experimental stage. At a minimum, you need macOS Catalina, Intel x86_64, XCode 11, and a Metal-compatible video card that meets the hardware requirements above.

## Linux 

> The primary Linux distribution for using the O3DE editor is Ubuntu 20.04.3 LTS. The following instructions will described how to retrieve and install the required software packages through Ubuntu's **apt-get** command-line utility.

Reviewer: ```suggestion
The primary Linux distribution for using the O3DE editor is Ubuntu 20.04.3 LTS. The following instructions describe how to retrieve and install the required software packages through Ubuntu's **apt-get** command-line utility.
```

### CMake

> As with the other operating systems, [CMake {{< versions/cmake >}} or later](https://cmake.org/download/) is required to configure and build O3DE projects. We strongly recommend that you install the **Latest Release** from the CMake rather than the default one provided by your current Linux distribution. If there is already a cmake installed, but does not match the minimum version, you will need to remove it with the following command.

Reviewer: ```suggestion
As with the other operating systems, [CMake {{< versions/cmake >}} or later](https://cmake.org/download/) is required to configure and build O3DE projects. We strongly recommend that you install the **Latest Release** of CMake rather than the default one provided by your current Linux distribution. If CMake is already installed, but does not match the minimum version, you will need to remove it with the following command.
```

```shell
sudo apt-get remove cmake
```

> In order to get the latest version of cmake for Ubuntu, you can add the Kitware APT repository to your ubuntu package list and run apt-get to install (See [Kitware APT Page](https://apt.kitware.com/) for details)

Reviewer: ```suggestion
In order to get the latest version of CMake for Ubuntu, you can add the Kitware APT repository to your Ubuntu package list and run **apt-get** to install it. Refer to [Kitware APT Page](https://apt.kitware.com/) for more information.
```

```shell
wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/null | gpg --dearmor - | sudo tee /usr/share/keyrings/kitware-archive-keyring.gpg >/dev/null

echo 'deb [signed-by=/usr/share/keyrings/kitware-archive-keyring.gpg] https://apt.kitware.com/ubuntu/ focal main' | sudo tee /etc/apt/sources.list.d/kitware.list >/dev/null

sudo apt-get update

sudo apt-get install cmake
```

Once installed, verify that the version meets the minimum version criteria.

```shell
cmake --version
```

### Clang

O3DE requires Clang to compile all of the native C++ code. Although Clang-6 is supported, it is recommended to use Clang-12 for the build process.

```shell
sudo apt-get install clang-12 
```

### Python 3.7 dependency on libffi

> Ubuntu 20.04.3 LTS does not support Python 3.7, however O3DE's local python package is Python 3.7 and has a library dependency on an earlier version of libffi that does not come with Ubunutu 20.04.3 LTS. You will need to manually install the older version of libffi onto Ubunutu 20.04.3 LTS in order for O3DE's Python to execute properly. The steps below describe create a temp folder under /tmp to download and manually install the specific debian package.

Reviewer: ```suggestion
Ubuntu 20.04.3 LTS does not support Python 3.7, however O3DE's local python package is Python 3.7 and has a library dependency on an earlier version of libffi that does not come with Ubunutu 20.04.3 LTS. You will need to manually install the older version of libffi onto Ubunutu 20.04.3 LTS in order for O3DE's Python to execute properly. The steps below demonstrate how to create a temp folder under `/tmp` to download and manually install the specific debian package.
```

> Ubuntu 20.04.3 LTS does not support Python 3.7, however O3DE's local python package is Python 3.7 and has a library dependency on an earlier version of libffi that does not come with Ubunutu 20.04.3 LTS. You will need to manually install the older version of libffi onto Ubunutu 20.04.3 LTS in order for O3DE's Python to execute properly. The steps below demonstrate how to create a temp folder under `/tmp` to download and manually install the specific debian package.

Reviewer 2: - I suggest adding the version of libffi that users should install. 
- Rephrased the first sentence for word flow and clarity. 
- Added a link to libffi, so users know what about the software they need to install.

```suggestion
O3DE's local python package, Python 3.7, depends on an earlier version of [libffi](https://sourceware.org/libffi/), which Ubuntu 20.04.3 LTS does not support. You will need to manually install an older version of libffi onto Ubuntu 20.04.3 LTS in order for O3DE's Python to run properly. The steps below demonstrate how to create a temp folder under `/tmp` to download and manually install the specific debian package for libffi. 
```

{{< note >}}

> Curl is required to perform the download step described in the instructions below.

Reviewer: ```suggestion
Curl is required to perform the download step shown in the following instructions.
```

> Curl is required to perform the download step shown in the following instructions.

Reviewer 2: - Switch to active voice. This gives users clear instruction of what action to take. 
```suggestion
You must install [Curl](https://github.com/curl/curl) to perform the download step shown in the following instructions.
```

Author: The URL referenced in this suggestion is for the curl open source. Typically in Linux, a package like Curl is installed with the linux distribution's package manager. For ubuntu, it would be :
```
sudo apt-get install curl
```
I can either remove the URL or add curl in the linux requirements in the welcome page (there is already a list of packages there)

{{< /note >}}

```shell
pushd /tmp

LIBFFI_PACKAGE_NAME=libffi6_3.2.1-8_amd64.deb
LIBFFI_PACKAGE_URL=http://mirrors.kernel.org/ubuntu/pool/main/libf/libffi/

curl --location $LIBFFI_PACKAGE_URL/$LIBFFI_PACKAGE_NAME -o $LIBFFI_PACKAGE_NAME

sudo apt install ./$LIBFFI_PACKAGE_NAME

popd
```

> ### Additional Library Dependencies

Reviewer 2: - Use sentence casing for headers, per O3DE style guide. 
```suggestion
### Additional library dependencies
```

> O3DE requires some additional library packages to be installed as well. 

Reviewer: ```suggestion
O3DE also requires some additional library packages to be installed.
```

> O3DE also requires some additional library packages to be installed.

Reviewer 2: ```suggestion
O3DE also requires some additional library packages to be installed:
```

* libglu1-mesa-dev
* libxcb-xinerama0
* libxcb-xinput0
* libxcb-xkb-dev
* libxkbcommon-dev
* libxkbcommon-x11-dev
* libfontconfig1-dev
* libcurl4-openssl-dev                    
* libsdl2-dev                             
* zlib1g-dev
* mesa-common-dev

> These packages can be downloaded and installed through **apt-get**.

Reviewer 2: ```suggestion
These packages can be downloaded and installed through `apt-get`.
```

```shell

> sudo apt-get install libglu1-mesa-dev libxcb-xinerama0 libxcb-xinput0 libfontconfig1-dev libcurl4-openssl-dev libsdl2-dev zlib1g-dev zlib1g-dev mesa-common-dev

Reviewer: You're missing a few packages in the apt-get install command that were in the list above it:

* libxcb-xkb-dev
* libxkbcommon-dev
* libxkbcommon-x11-dev

Author: good catch

```

### Ninja Build System (Optional)

> CMake supports Linux's default 'Unix Makefiles'. However, this does not support multiple configurations, so when a cmake project is generated, you must supply which configuration you want the build for (debug, profile, or release). The Ninja build system, specifically the [Ninja Multi-Confg](https://cmake.org/cmake/help/latest/generator/Ninja%20Multi-Config.html) does support multiple configurations within the same generated build folder, so it is recommended to use this generator for O3DE development. This can be installed through **apt-get**

Reviewer: ```suggestion
CMake supports Linux's default 'Unix Makefiles'. However, this does not support multiple configurations, so when a CMake project is generated, you must supply the configuration you want to build (debug, profile, or release). The Ninja build system, specifically [Ninja Multi-Config](https://cmake.org/cmake/help/latest/generator/Ninja%20Multi-Config.html), supports multiple configurations within the same generated build folder, so we recommend that you use this generator for O3DE development. Ninja Multi-Config can be installed through **apt-get**:
```

Reviewer 3: I like this re-wording, except for the very last one. It is not the "Ninja Multi-Config" that is installed with `apt install ninja-build`. "Ninja Multi-Config" is the name of the CMake generator, and is part of CMake itself. `apt install ninja-build` installs the "Ninja build tool".

Author: Will update to 
```
Ninja build tool can be installed through **apt-get**:
```

> CMake supports Linux's default 'Unix Makefiles'. However, this does not support multiple configurations, so when a CMake project is generated, you must supply the configuration you want to build (debug, profile, or release). The Ninja build system, specifically [Ninja Multi-Config](https://cmake.org/cmake/help/latest/generator/Ninja%20Multi-Config.html), supports multiple configurations within the same generated build folder, so we recommend that you use this generator for O3DE development. Ninja build tool can be installed through **apt-get**:

Reviewer 2: - It wasn't clear to me the relationship between Unix Makefiles and Ninja Multi-Config. I suggest rephrasing it to more clearly explain why they would need Ninja Build System, especially for those who aren't familiar. 
- Also, I'm not sure when you need to supply the configuration (debug, profile, or release)? Is this part necessary in this section?

```suggestion
CMake readily supports Linux's default Unix Makefiles; however, Unix Makefiles only supports single-configuration builds and not multi-configuration builds. The Ninja build system, specifically [Ninja Multi-Config](https://cmake.org/cmake/help/latest/generator/Ninja%20Multi-Config.html), supports multiple configurations within the same generated build folder. We recommend that you use this generator for O3DE development. You can install Ninja built tool through `apt-get`. 
```

Author: I will re-word this section to give more clarity
